plugins/shortcuts/src/AddShortcut.tsx

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import { useLocation } from 'react-router';
+import { SubmitHandler } from 'react-hook-form';
+import { alertApiRef, useApi } from '@backstage/core';
+import {
+  Button,
+  Card,
+  CardHeader,
+  makeStyles,
+  Popover,
+} from '@material-ui/core';
+import { ShortcutForm } from './ShortcutForm';
+import { FormValues, Shortcut } from './types';
+import { ShortcutApi } from './api';
+
+const useStyles = makeStyles(theme => ({
+  card: {
+    width: 400,
+  },
+  header: {
+    marginBottom: theme.spacing(1),
+  },
+  button: {
+    marginTop: theme.spacing(1),
+  },
+}));
+
+type Props = {
+  onClose: () => void;
+  anchorEl?: Element;
+  api: ShortcutApi;
+};
+
+export const AddShortcut = ({ onClose, anchorEl, api }: Props) => {
+  const classes = useStyles();
+  const alertApi = useApi(alertApiRef);
+  const { pathname } = useLocation();
+  const [formValues, setFormValues] = useState<FormValues>();
+  const open = Boolean(anchorEl);
+
+  const handleSave: SubmitHandler<FormValues> = async ({ url, title }) => {
+    const shortcut: Omit<Shortcut, 'id'> = { url, title };
+
+    try {
+      await api.add(shortcut);
+      alertApi.post({
+        message: 'Successfully added shortcut',
+        severity: 'success',
+      });
+    } catch (error) {
+      alertApi.post({
+        message: `Could not add shortcut: ${error.message}`,
+        severity: 'error',
+      });
+    }
+
+    onClose();
+  };
+
+  const handlePaste = () => {
+    setFormValues({ url: pathname, title: document.title });
+  };
+
+  const handleClose = () => {
+    setFormValues(undefined);
+    onClose();
+  };
+
+  return (
+    <Popover
+      open={open}
+      anchorEl={anchorEl}
+      onExit={handleClose}
+      onClose={onClose}
+      anchorOrigin={{
+        vertical: 'top',
+        horizontal: 'right',
+      }}
+    >
+      <Card className={classes.card}>
+        <CardHeader
+          className={classes.header}
+          title="Add Shortcut"
+          titleTypographyProps={{ variant: 'subtitle2' }}
+          action={
+            <Button
+              className={classes.button}
+              variant="text"
+              size="small"
+              color="primary"
+              onClick={handlePaste}
+            >
+              Paste Current Url
```

Comment:
```suggestion
              Paste Current URL
```

As an acronym, think this reference of Universal Resource Locater here should be all caps maybe?